xarray/tests/test_plot.py

```diff
@@ -1456,6 +1456,50 @@ def test_default_labels(self):
             assert substring_in_axes(label, ax)
 
 
+class TestFacetedLinePlots(PlotTestCase):
+    def setUp(self):
+        self.darray = DataArray(np.random.randn(10, 6, 3, 4),
+                                dims=['hue', 'x', 'col', 'row'],
+                                coords=[range(10), range(6),
+                                        range(3), ['A', 'B', 'C', 'C++']],
+                                name='Cornelius Ortega the 1st')
+
+    def test_facetgrid_shape(self):
+        g = self.darray.plot(row='row', col='col', hue='hue')
+        assert g.axes.shape == (len(self.darray.row), len(self.darray.col))
+
+        g = self.darray.plot(row='col', col='row', hue='hue')
+        assert g.axes.shape == (len(self.darray.col), len(self.darray.row))
+
+    def test_default_labels(self):
+        g = self.darray.plot(row='row', col='col', hue='hue')
+        # Rightmost column should be labeled
+        for label, ax in zip(self.darray.coords['row'].values, g.axes[:, -1]):
+            assert substring_in_axes(label, ax)
+
+        # Top row should be labeled
+        for label, ax in zip(self.darray.coords['col'].values, g.axes[0, :]):
+            assert substring_in_axes(str(label), ax)
+
+        # Leftmost column should have array name
+        for ax in g.axes[:, 0]:
+            assert substring_in_axes(self.darray.name, ax)
+
+    def test_test_empty_cell(self):
+        g = self.darray.isel(row=1).drop('row').plot(col='col',
+                                                     hue='hue',
+                                                     col_wrap=2)
+        bottomright = g.axes[-1, -1]
+        assert not bottomright.has_data()
+        assert not bottomright.get_visible()
+
+    def test_set_axis_labels(self):
+        g = self.darray.plot(row='row', col='col', hue='hue')
+        g.set_axis_labels('longitude', 'latitude')
+        alltxt = text_in_fig()
+        for label in ['longitude', 'latitude']:
+            assert label in alltxt
```

Comment:
nit: for unit tests, we prefer to write things out in long form when practical rather than writing loops. This makes things easier to debug, because test runners usually only report the line on which the error occurs, e.g.,
```
assert 'longitude' in alltxt
assert 'latitude' in alltxt
```